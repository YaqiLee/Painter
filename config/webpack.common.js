const path = require("path");
const { CleanWebpackPlugin } = require("clean-webpack-plugin");
const HtmlWebpackPlugin = require("html-webpack-plugin");

module.exports = {
  entry: "./src/index.js",
  output: {
    filename: "[name].js",
    path: path.resolve(__dirname, "../", "dist"),
  },

  module: {
    rules: [
      {
        test: /\.jsx?$/,
        loader: "babel-loader",
        exclude: /node_modules/
      },
      {
        test: /\.scss$/,
        use: [
          // Creates `style` nodes from JS strings
          "style-loader",
          // Translates CSS into CommonJS
          "css-loader",
          // Compiles Sass to CSS
          "sass-loader",
        ],
      },
      {
        test: /\.(png|svg|jpg|jpeg|gif)$/i,
        type: "asset/resource",
      },
    ],
  },
  plugins: [
    new CleanWebpackPlugin(),
    new HtmlWebpackPlugin({
      template: "./src/index.html",
      title: "Painter",
    }),
  ],
};
