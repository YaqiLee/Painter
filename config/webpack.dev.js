const { merge } = require("webpack-merge");
const common = require("./webpack.common.js");

module.exports = merge(common, {
  mode: "development",

  devtool: "inline-source-map",
  cache: false,

  devServer: {
    port: 4600
  },
});
