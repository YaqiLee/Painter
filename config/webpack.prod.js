const { merge } = require("webpack-merge");
const common = require("./webpack.common.js");
const webpack = require("webpack");
const MiniCssExtractPlugin = require('mini-css-extract-plugin');

module.exports = merge(common, {
  bail: true,
  mode: "production",
  devtool: "source-map", // 可以在生产环境中启用source-map, 可以追踪源码
  optimization: {
    splitChunks: {
      chunks: "all",
      name: "common"
    }
  },
  stats: {
    assets: true
  },
  module: {
    rules: [
      {
        test: /\.scss$/i,
        use: [MiniCssExtractPlugin.loader, 'css-loader', 'sass-loader'],
      },
    ],
  },
  plugins: [
    new MiniCssExtractPlugin(),
    new webpack.DefinePlugin({
      "process.env.NODE_ENV": JSON.stringify("production"),
    }),
  ]
});
