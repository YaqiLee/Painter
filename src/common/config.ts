export const brush = {
  select: "select", // 选区
  curve: "curve",
  line: "line",
  rect: "rect", // 填充
  oRect: "oRect", // 空心矩形
  circle: "circle", // 填充圆
  oCircle: "oCircle", // 空心圆
};

export enum BrushShape {
  select,
  curve,
  line,
  rect,
  circle,
}

export const KEY = {
  SHIFT: "Shift",
  DELETE: "Delete",
};
