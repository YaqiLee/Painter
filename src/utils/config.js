export const brush = {
  curve: 'curve',
  line: 'line',
  rect: 'rect', // 填充
  oRect: 'oRect', // 空心矩形
  circle: 'circle', // 填充圆
  oCircle: 'oCircle', // 空心圆
}
