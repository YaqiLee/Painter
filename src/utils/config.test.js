import { brush } from "./config";

test("笔刷类型不要改变", () => {
  expect(brush).toMatchSnapshot();
})