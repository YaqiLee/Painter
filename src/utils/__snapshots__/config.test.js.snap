// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`笔刷类型不要改变 1`] = `
Object {
  "circle": "circle",
  "curve": "curve",
  "line": "line",
  "oCircle": "oCircle",
  "oRect": "oRect",
  "rect": "rect",
  "select": "select",
}
`;

exports[`配置文件变动 1`] = `
Object {
  "circle": "circle",
  "curve": "curve",
  "line": "line",
  "oCircle": "oCircle",
  "oRect": "oRect",
  "rect": "rect",
  "select": "select",
}
`;
